streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -580,4 +562,118 @@ private String getBrokerSideConfigValue(final Config brokerSideTopicConfig,
 
         return topicsToCreate;
     }
+
+    /**
+     * Sets up internal topics.
+     *
+     * Either the given topic are all created or the method fails with an exception.
+     *
+     * @param topicConfigs internal topics to setup
+     */
+    public void setup(final Map<String, InternalTopicConfig> topicConfigs) {
+        log.info("Starting to setup internal topics {}.", topicConfigs.keySet());
+
+        final long now = time.milliseconds();
+        final long deadline = now + retryTimeoutMs;
+
+        final Map<String, Map<String, String>> newTopicConfigs = topicConfigs.values().stream()
```

Comment:
naming nit: topicConfigsWithRetention

Reply:
I would not rename this since the additional retention is only added in case of `topicConfig instanceof WindowedChangelogTopicConfig`, which is one of three. What about `streamsSideTopicConfigs` to emphasize that these are the topic configs that Streams sets by default possibly overidden by user code in the Streams app.